CHANGELOG.md

```diff
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to
 [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## 17.0.2
```

Comment:
Please don't do this.

Document versioned releases in the CHANGELOG when they're real. Accumulate un-released change in a next-version placeholder heading until the change is released.

[Looking at Maven central](https://search.maven.org/artifact/org.apereo.uportal/uportal-app-framework), looks like the most recent release is 16.0.2. So the most recent named versioned release in the CHANGELOG should be 16.0.2, and that heading should document what changes were included in 16.0.2.  All changes since then should accumulate in a next-version heading, awaiting release.

*Not* automatically issuing version numbers for each pull request is a good thing. Or at least it's a way to navigate the tradeoffs that has some upside. The major upside is that changes can accumulate into a semantic whole, tested in predev, before shipping a release. Taking multiple PRs to get a release to stable, using the feedback of seeing the latest unreleased on predev, is okay.

Hoping this clarifies. I'm game to talk through release strategy and tactics over video conference if that would help. I acknowledge that different MyUW-related products have different release practices and that this makes matching the practices to the contexts more complex. Nonetheless, local practice in this repo is to decouple Pull Request merges from releases.